src/demangle.cc

```diff
@@ -223,9 +223,6 @@ static bool ParseTwoCharToken(State *state, const char *two_char_token) {
 // Returns true and advances "mangled_cur" if we find any character in
 // "char_class" at "mangled_cur" position.
 static bool ParseCharClass(State *state, const char *char_class) {
-  if (state->mangled_cur == '\0') {
```

Comment:
Why are we removing this code?

Reply:
I have removed it based on this commit find it on the upstream: https://github.com/google/glog/commit/a5ffa884137f7687d0393ccba22557d583654a25#diff-81af9f0eab9a04e1d26013a0af666c73

Reply:
I see, then it should be equivalent code to only have the for loop as it won't match a '\0'.